UI/window-basic-main.cpp

```diff
@@ -1596,6 +1612,207 @@ void OBSBasic::DeferredLoad(const QString &file, int requeueCount)
 	Load(QT_TO_UTF8(file));
 }
 
+void OBSBasic::SaveScreenshotHandler()
+{
+	const bool noSpace = true;
+
+	// Calculate output file path
+	std::string strOutputFilePath =
+		config_get_string(Config(), "ScreenShots", "FilePath");
+
+	char lastChar = strOutputFilePath.back();
+	if (lastChar != '/' && lastChar != '\\')
+		strOutputFilePath += "/";
+
+	// TODO: Take this out to configuration
+	strOutputFilePath += GenerateSpecifiedFilename("png", noSpace, "%CCYY-%MM-%DD %hh-%mm-%ss");
+
+	replace(strOutputFilePath.begin(), strOutputFilePath.end(), '\\', '/');
+
+	// Get & ensure directory exists
+	size_t last = strOutputFilePath.rfind('/');
+	if (last == string::npos)
+		return;
+
+	string directory = strOutputFilePath.substr(0, last);
+	os_mkdirs(directory.c_str());
+
+	// Save screenshot to specified file path
+	bool success = SaveScreenshotToFile(strOutputFilePath.c_str());
+
+	if (success)
+	{
+		// TODO: Audio/visual feedback
+	}
+}
+
+bool OBSBasic::SaveScreenshotToFile(const char* outputFilePath)
```

Comment:
This function is … really big; you should consider splitting it up, and maybe even moving (all) the parts to a separate file (e.g. not have any ffmpeg includes in UI/window-basic-main.cpp)